src/tests/integration/SparkLend.t.sol

```diff
@@ -0,0 +1,400 @@
+        hub.exec(ilk);
+
+        // User repays half their debt
+        assertEq(getDebt(), buffer + buffer / 2);
+        sparkPool.repay(address(dai), buffer / 4, 2, address(this));
+        assertEq(getDebt(), buffer + buffer / 2);
+
+        hub.exec(ilk);
+
+        assertEq(getDebt(), buffer + buffer / 2 - buffer / 4, "should be back down to 1.25x the buffer");
+    }
+
+    /** 
+     * The DAI market is using a new interest model which over-allocates interest to the treasury.
+     * This is due to the reserve factor not being flexible enough to account for this.
+     * Confirm that we can later correct the discrepancy by donating the excess liabilities back to the DAI pool. (This can be automated later on)
+     */
+    function test_asset_liabilities_fix() public {
+        uint256 assets = getTotalAssets(address(dai));
+        uint256 liabilities = getTotalLiabilities(address(dai));
+        if (assets >= liabilities) {
+            // Force the assets to become less than the liabilities
+            uint256 performanceBonus = daiInterestRateStrategy.performanceBonus();
+            vm.prank(admin); plan.file("buffer", performanceBonus * 4);
+            hub.exec(ilk);
+            sparkPool.borrow(address(dai), performanceBonus * 2, 2, 0, address(this));  // Supply rate should now be above 0% (we are over-allocating)
+
+            // Warp so we gaurantee there is new interest
+            vm.warp(block.timestamp + 365 days);
+            forceUpdateIndicies(address(dai));
+
+            assets = getTotalAssets(address(dai));
+            liabilities = getTotalLiabilities(address(dai));
+            assertLe(assets, liabilities, "assets should be less than or equal to liabilities");
+        }
+        
+        // Let's fix the accounting
+        uint256 delta = liabilities - assets;
+
+        // First trigger all spDAI owed to the treasury to be accrued
+        assertGt(getAccruedToTreasury(address(dai)), 0, "accrued to treasury should be greater than 0");
+        address[] memory toMint = new address[](1);
+        toMint[0] = address(dai);
+        sparkPool.mintToTreasury(toMint);
+        assertEq(getAccruedToTreasury(address(dai)), 0, "accrued to treasury should be 0");
+        assertGe(adai.balanceOf(address(treasury)), delta, "adai treasury should have more than the delta between liabilities and assets");
+
+        // Donate the excess liabilities back to the pool
+        // This will burn the liabilities while keeping the assets the same
+        vm.prank(admin); treasuryAdmin.transfer(address(treasury), address(adai), address(this), delta);
+        sparkPool.withdraw(address(dai), delta, address(adai));
+
+        assets = getTotalAssets(address(dai)) + 1;  // In case of rounding error we +1
+        liabilities = getTotalLiabilities(address(dai));
+        assertGe(assets, liabilities, "assets should be greater than or equal to liabilities");
+    }
+}
```

Comment:
FYI I think `test_collect_stkaave` and `test_collect_stkaave_king_not_set` do not have counterparts here.